browser/extensions/api/brave_rewards_api.cc

```diff
@@ -27,6 +33,32 @@ ExtensionFunction::ResponseAction BraveRewardsCreateWalletFunction::Run() {
   return RespondNow(NoArguments());
 }
 
+BraveRewardsDonateToSiteFunction::~BraveRewardsDonateToSiteFunction() {
+}
+
+ExtensionFunction::ResponseAction BraveRewardsDonateToSiteFunction::Run() {
+  std::unique_ptr<brave_rewards::DonateToSite::Params> params(
+      brave_rewards::DonateToSite::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  // Get web contents for this tab
+  content::WebContents* contents = nullptr;
+  if (!ExtensionTabUtil::GetTabById(
```

Comment:
- [ ] `BraveRewardsDonations` should not be available in tor-tabs. Verify if the tab is `tor tab` before opening the donation dialog.

Reply:
@bridiver The Rewards popup does not use WebUI, it's an extension. This API method gives the extension an way to open the donate WebUI in a dialog.

@jumde Is it neccessary to block Tor WebContents here given that the extension popup (and therefore button which calls this API) will be disabled for Tor tabs via https://github.com/brave/brave-browser/issues/1484?

Reply:
@petemill - https://github.com/brave/brave-browser/issues/1484 disables the BAT icon. But navigation to `chrome://donate` is still possible in tor-mode. 

<img width="1435" alt="screen shot 2018-10-11 at 12 37 22 pm" src="https://user-images.githubusercontent.com/1903815/46829370-72182f00-cd52-11e8-95a2-e536ba992d6f.png">

Reply:
Sure but the UI is not provided any publisher information, so it won't do anything @jumde

Reply:
@jumde just pushed disabling the content in private / tor contexts

![image](https://user-images.githubusercontent.com/741836/46830162-b6a4ca00-cd54-11e8-8c57-b8f6e43b36c3.png)

Also makes sure Rewards service never starts in tor context too, previously only prevented in private context.